src/lib_ccx/ccx_encoders_webvtt.c

```diff
@@ -214,9 +214,19 @@ int write_webvtt_header(struct encoder_ctx *context)
 		int used;
 		unsigned h1, m1, s1, ms1;
 		millis_to_time(context->timing->sync_pts2fts_fts, &h1, &m1, &s1, &ms1);
-		sprintf(header_string, "X-TIMESTAMP-MAP=MPEGTS:%ld,LOCAL:%02u:%02u:%02u.%03u%s",
-			context->timing->sync_pts2fts_pts, h1, m1, s1, ms1,
-			ccx_options.enc_cfg.line_terminator_lf ? "\n\n" : "\r\n\r\n");
+		 
+		 // If the user has not disabled the X-TIMESTAMP-MAP header
+		if (!ccx_options.no_timestamp_map)
```

Comment:
Encoders should get options from their context, not directly from the user parameters. 
So you need to copy from the user options to the encoder context (you'll see plenty of examples of this in the code, and I believe the tutorial we have about this about does it).

Reply:
I added the change!